app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,15 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_post, only: [:show, :edit, :update, :destroy, :favorite, :unfavorite]
+  before_action :block_unauthorized_changes, only: [:edit, :update, :destroy]
   before_filter :authenticate_user!
   # GET /posts
-  # GET /posts.json
   def index
-    @posts = Post.all.includes(:user)
+    # @posts = Post.all.includes(:user)
+    @posts = Post.where(user_id: current_user.following.map(&:id)).includes(:user)
```

Comment:
Yes, I was conscious about having too many queries on a page. I did not know about the ```.ids``` trick. That is pretty cool. I'll definitely use that.